### ISSUE TITLE
[rush] Consolidate `rush update` and `rush update-autoinstaller`

### ISSUE THREAD

**Issue by isc30** [microsoft/rushstack#3230]
Would you be happy if I create a PR that makes `rush update` also update autoinstallers?

I've found that many devs get used to `rush update` and forget about the autoinstallers, so this could abstract them from knowing the difference.

Please let me know what you think!

**Comment by isc30** [comment]
Another improvement could be also installing the autoinstallers when doing a `rush install`.

In our case, we have a `prettier` autoinstaller configured with some plugins, and these plugins cannot be found by vscode until we do the first commit that runs prettier and installs them.

**Comment by D4N14L** [comment]
Hmm... The purpose of the auto-installer is generally intended to lazy-install required deps only on an as-needed basis, and reduce install time/overhead. If you required the packages to be consistently updated, you could instead make the project a member of the core repository which would allow the package to be updated when `rush update` is run. Does this sound like it fits your needs?

**Comment by isc30** [comment]
Yeah I think that would work well for the `rush install` case where we don't want to install the dependencies for autoinstallers and make it lazy.

In terms of having many people working on the repo, I find that `rush update` could also update autoinstallers and it shouldn't affect the results in any negative way:

- autoisntallers package.json hasn't changed = pnpm install will do nothing (fast)
- autoinstallers have package.json changed = rush will update them and get proper lockfiles etc (time saver for dev and prevents forgetting about it)

In our case, we have converted these to be normal projects that get built/installed with `rush install` / `rush update` but maybe `rush update` could check if the autoinstaller `package.json` changed and simplify that?

thanks!

**Comment by isc30** [comment]
> Hmm... The purpose of the auto-installer is generally intended to lazy-install required deps only on an as-needed basis, and reduce install time/overhead. If you required the packages to be consistently updated, you could instead make the project a member of the core repository which would allow the package to be updated when `rush update` is run. Does this sound like it fits your needs?

Hi again, I'm running into an issue when trying to make the packages be normal packages rather than autoinstallers: I can't use them when running commands from `command-line.json`

**Comment by isc30** [comment]
adding a comment here with the workaround as some people requested it:

`common/autoinstallers/install.js`
```js
const execSync = require('child_process').execSync
const autoinstallers = ['rush-prettier'] // add more if needed

autoinstallers.forEach(a => {
  execSync('pnpm install', { cwd: `${__dirname}/${a}`, stdio: 'inherit' })
})
```

`rush.json`:
```json
"eventHooks": {
  "postRushInstall": ["node common/autoinstallers/install.js"],
}
```

**Comment by elliot-nelson** [comment]
Hi @isc30, if you're still interested, I'd like to get your take on the proposed changes in PR https://github.com/microsoft/rushstack/pull/3789.

It's not quite the same thing because the new `rush install --autoinstallers` command is an optional thing (the user has to opt-in, rather than it happening automatically).

Is that sufficient for your needs, or do you really need it to happen on every rush install?

**Comment by isc30** [comment]
hey @elliot-nelson the PR seems like a step in the good direction. I was thinking on getting autoinstallers updated automatically on `rush update` since its fast (no operation) if nothing changed there
